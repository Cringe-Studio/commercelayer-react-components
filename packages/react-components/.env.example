# Unit tests
VITE_TEST_CLIENT_ID=
VITE_TEST_ENDPOINT=
VITE_TEST_MARKET_ID=
VITE_TEST_USERNAME=
VITE_TEST_PASSWORD=
VITE_TEST_CLIENT_ID_INTEGRATION=
VITE_TEST_CLIENT_SECRET=
