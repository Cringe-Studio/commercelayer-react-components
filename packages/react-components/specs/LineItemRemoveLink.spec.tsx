import React from 'react'
import { LineItemRemoveLink } from '../packages/react-components/src'
import renderer from 'react-test-renderer'
import components from '../#config/components'
import Parent from '../#components/utils/Parent'

const propTypes = components.LineItemRemoveLink.propTypes

test('<LineItemRemoveLink/>', () => {
  expect.assertions(3)
  const component = renderer.create(<LineItemRemoveLink />)
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
  expect(proptypes.label).toBe(propTypes.label)
})

test('<LineItemRemoveLink check children format />', () => {
  expect.assertions(3)
  const component = renderer.create(
    <LineItemRemoveLink label="remove product" />
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.type).toBe('a')
  expect(root.props.label).toBe('remove product')
})

test('<LineItemRemoveLink with custom children />', () => {
  expect.assertions(7)
  const CustomComponent = (props) => <span>{props.label}</span>
  const component = renderer.create(
    <LineItemRemoveLink>{CustomComponent}</LineItemRemoveLink>
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const childRendered = root.rendered.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.type).toBe(Parent)
  expect(rendered.props.handleRemove).toBeDefined()
  expect(childRendered.nodeType).toBe('component')
  expect(childRendered.type).toBe(CustomComponent)
  expect(childRendered.rendered.nodeType).toBe('host')
  expect(childRendered.rendered.type).toBe('span')
})
