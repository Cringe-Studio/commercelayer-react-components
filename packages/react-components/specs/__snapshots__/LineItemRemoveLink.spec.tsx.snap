// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineItemRemoveLink check children format /> 1`] = `
<a
  href="#"
  label="remove product"
  onClick={[Function]}
>
  remove product
</a>
`;

exports[`<LineItemRemoveLink with custom children /> 1`] = `
<span>
  Remove
</span>
`;

exports[`<LineItemRemoveLink/> 1`] = `
<a
  href="#"
  label="Remove"
  onClick={[Function]}
>
  Remove
</a>
`;
